tests/suites/test_suite_psa_crypto.function

```diff
@@ -8779,54 +8791,90 @@ void ecjpake_setup( int alg_arg, int key_type_pw_arg, int key_usage_pw_arg,
 
     PSA_ASSERT( psa_pake_abort( &operation ) );
 
-    TEST_EQUAL( psa_pake_set_user( &operation, NULL, 0 ),
-                PSA_ERROR_BAD_STATE );
-    TEST_EQUAL( psa_pake_set_peer( &operation, NULL, 0 ),
-                PSA_ERROR_BAD_STATE );
-    TEST_EQUAL( psa_pake_set_password_key( &operation, key ),
-                PSA_ERROR_BAD_STATE );
-    TEST_EQUAL( psa_pake_set_role( &operation, role ),
-                PSA_ERROR_BAD_STATE );
-    TEST_EQUAL( psa_pake_output( &operation, PSA_PAKE_STEP_KEY_SHARE,
-                                 NULL, 0, NULL ),
-                PSA_ERROR_BAD_STATE );
-    TEST_EQUAL( psa_pake_input( &operation, PSA_PAKE_STEP_KEY_SHARE, NULL, 0),
-                PSA_ERROR_BAD_STATE );
-
-    PSA_ASSERT( psa_pake_abort( &operation ) );
+    if ( inj_err_type == INJECT_ERR_UNINITIALIZED_ACCESS )
+    {
+        TEST_EQUAL( psa_pake_set_user( &operation, NULL, 0 ),
+                    expected_error );
+        PSA_ASSERT( psa_pake_abort( &operation ) );
+        TEST_EQUAL( psa_pake_set_peer( &operation, NULL, 0 ),
+                    expected_error );
+        PSA_ASSERT( psa_pake_abort( &operation ) );
+        TEST_EQUAL( psa_pake_set_password_key( &operation, key ),
+                    expected_error );
+        PSA_ASSERT( psa_pake_abort( &operation ) );
+        TEST_EQUAL( psa_pake_set_role( &operation, role ),
+                    expected_error );
+        PSA_ASSERT( psa_pake_abort( &operation ) );
+        TEST_EQUAL( psa_pake_output( &operation, PSA_PAKE_STEP_KEY_SHARE,
+                    NULL, 0, NULL ),
+                    expected_error );
+        PSA_ASSERT( psa_pake_abort( &operation ) );
+        TEST_EQUAL( psa_pake_input( &operation, PSA_PAKE_STEP_KEY_SHARE, NULL, 0),
+                    expected_error );
+        PSA_ASSERT( psa_pake_abort( &operation ) );
+        goto exit;
+    }
 
-    TEST_EQUAL( psa_pake_setup( &operation, &cipher_suite ),
-                expected_status_setup );
-    if( expected_status_setup != PSA_SUCCESS )
+    status = psa_pake_setup( &operation, &cipher_suite );
+    if (status != PSA_SUCCESS)
```

Comment:
This doesn't look correct: basically we're asserting that `status` is either success or the expected error. But we should know whether we expect success or not at this stage and assert that. I mean currently, imagine we invoke the function with `INJECT_ERR_DUPLICATE_SETUP, PSA_ERROR_BAD_STATE` as the last two arguments. The `psa_pake_setup()` is expected to return success, but currently if it returns `PSA_ERROR_BAD_STATE` for some reason the test will still be marked as passed.

The underlying issue is that we have a single `expected_error` variable for all functions that may fail. We should either have multiple variables (`expected_status_setup`, `expected_status_set_role`, etc) or an extra variable `expecting_error_at_stage_xxx`, taking value from a new enum, or any other way to ensure that only the function that we expect to fail actually fails.

Reply:
Actually perhaps we can extend the existing enum use for injecting "internal" errors to indicate the stage for "external" errors as well:
```c
typedef enum
{
    ERR_NONE = 0,
    ERR_INJECT_UNINITIALIZED_ACCESS,
    /* ... */
    ERR_IN_SETUP,
    ERR_IN_SET_PASSWORD_KEY,
    /* ... */
} ecjpake_error_stage_t;
```
then when we pass arguments that will cause setup to fail, we also pass `ERR_IN_SETUP`, etc.

And here we would do
```c
if( error_stage != ERR_IN_SETUP )
{
    PSA_ASSERT( status );
}
else
{
    TEST_EQUAL( status, expected_error );
    goto exit;
}
```

Wdyt?